rotkehlchen/db/search_assets.py

```diff
@@ -0,0 +1,134 @@
+    with GlobalDBHandler().conn.read_ctx() as globaldb_cursor:
+        globaldb_cursor.execute(query, bindings)
+        treat_eth2_as_eth = db.get_settings(cursor).treat_eth2_as_eth
+        found_eth = False
+        for entry in globaldb_cursor:
+            # default to a value greater than the threshold to avoid false-positives
+            # and prevent raising `UnboundLocalError` when name/symbol is None.
+            lev_dist_min = 6
+            if entry[1] is not None:
+                lev_dist_min = min(
+                    lev_dist_min,
+                    levenshtein(substring_search, entry[1].casefold()),
+                )
+            if entry[2] is not None:
+                lev_dist_min = min(
+                    lev_dist_min,
+                    levenshtein(substring_search, entry[2].casefold()),
+                )
+            if treat_eth2_as_eth is True and entry[0] in (A_ETH.identifier, A_ETH2.identifier):  # noqa:E501
+                if found_eth is False:
+                    search_result.append((lev_dist_min, {
+                        'identifier': resolved_eth.identifier,
+                        'name': resolved_eth.name,
+                        'symbol': resolved_eth.symbol,
+                        'asset_type': 'own chain',
+                    }))
+                    found_eth = True
+                continue
+
+            entry_info = {
+                'identifier': entry[0],
+                'name': entry[1],
+                'symbol': entry[2],
+                'asset_type': AssetType.deserialize_from_db(entry[4]).serialize(),
+            }
+            if entry[3] is not None:
+                entry_info['evm_chain'] = ChainID.deserialize_from_db(entry[3]).serialize()
+            if entry[5] is not None:
+                entry_info['custom_asset_type'] = entry[5]
+
+            search_result.append((lev_dist_min, entry_info))
+
+    return search_result
+
+
+def search_assets_levenshtein(
+        db: 'DBHandler',
+        substring_search: str,
+        limit: Optional[int],
+        evm_chain: Optional[ChainID],
+        search_nfts: bool,
+) -> List[Dict[str, Any]]:
+    """Returns a list of asset details that match the search keyword using the Levenshtein distance approach."""  # noqa: E501
+    substring_search = substring_search.casefold()
+    search_result = []
+    with db.conn.read_ctx() as cursor:
+        search_result = _search_only_assets_levenstein(
+            cursor=cursor,
+            db=db,
+            substring_search=substring_search,
+            evm_chain=evm_chain,
+        )
+        if search_nfts is True:
+            search_result += _search_only_nfts_levenstein(
+                cursor=cursor,
+                substring_search=substring_search,
+            )
+
+    sorted_search_result = [result for _, result in sorted(search_result, key=lambda item: item[0])]  # noqa: E501
+    return sorted_search_result[:limit] if limit is not None else sorted_search_result
```

Comment:
For the record tried the old test (with the old logic) vs the new test (bigger) with the new logic and they take on average the same time (cold start, remember that the new one does more things) + the new one in a warm start is much faster (with more things tested)

Reply:
as I asked also Alexey. Did you include heavy NFT accounts and nft search in those tests?